mbedtls/src/pk/mod.rs

```diff
@@ -1064,6 +1088,135 @@ iy6KC991zzvaWY/Ys+q/84Afqa+0qJKQnPuy/7F5GkVdQA/lfbhi
         assert_ne!(&cipher1[..], &cipher2[..]);
     }
 
+    #[test]
+    fn encrypt_raw_decrypt_with_pkcs1_v15() {
+        let mut pk = Pk::from_private_key(TEST_DER, None).unwrap();
+        let mut cipher = [0u8; 2048 / 8];
+        let mut rng = crate::test_support::rand::test_rng();
+        pk.set_options(Options::Rsa {
+            padding: RsaPadding::Pkcs1V15
+        });
+        assert_eq!(
+            pk.encrypt(b"test", &mut cipher, &mut rng)
+                .unwrap(),
+            cipher.len()
+        );
+        let mut decrypted_data1 = [0u8; 2048 / 8];
+        let length_without_padding = pk.decrypt(&cipher, &mut decrypted_data1, &mut rng).unwrap();
+        // set raw decryption padding mode to perform raw decryption
+        pk.set_options(Options::Rsa {
+            padding: RsaPadding::None
+        });
+        let mut decrypted_data2 = [0u8; 2048 / 8];
+        let length_with_padding = pk.decrypt(&cipher, &mut decrypted_data2, &mut rng).unwrap();
+        // compare lengths of the decrypted texts
+        assert_ne!(length_without_padding, length_with_padding);
+        assert_ne!(&decrypted_data1[..], &decrypted_data2[..]);
+    }
+
+    #[test]
+    fn encrypt_raw_decrypt_with_pkcs1_v21() {
+        let mut pk = Pk::from_private_key(TEST_DER, None).unwrap();
+        let mut cipher = [0u8; 2048 / 8];
+        let mut rng = crate::test_support::rand::test_rng();
+        pk.set_options(Options::Rsa {
+            padding: RsaPadding::Pkcs1V21 {
+                mgf: Type::Sha256,
+            },
+        });
+        assert_eq!(
+            pk.encrypt(b"test", &mut cipher, &mut rng)
+                .unwrap(),
+            cipher.len()
+        );
+        let mut decrypted_data1 = [0u8; 2048 / 8];
+        let length_without_padding = pk.decrypt(&cipher, &mut decrypted_data1, &mut rng).unwrap();
+        // set raw decryption padding mode to perform raw decryption
+        pk.set_options(Options::Rsa {
+            padding: RsaPadding::None
+        });
+        let mut decrypted_data2 = [0u8; 2048 / 8];
+        let length_with_padding = pk.decrypt(&cipher, &mut decrypted_data2, &mut rng).unwrap();
+        // compare lengths of the decrypted texts
+        assert_ne!(length_without_padding, length_with_padding);
+        assert_ne!(&decrypted_data1[..], &decrypted_data2[..]);
+    }
+
+    #[test]
+    fn rsa_encrypt_with_no_padding() {
+        let mut pk = Pk::from_private_key(TEST_DER, None).unwrap();
+        let mut cipher = [0u8; 2048 / 8];
+        pk.set_options(Options::Rsa {
+            padding: RsaPadding::None
+        });
+        assert!(pk.encrypt(b"test", &mut cipher, &mut crate::test_support::rand::test_rng()).is_err());
```

Comment:
In general: use `result.unwrap_err()` instead of `assert!(result.is_err())`. In this case, please also check for the specific error code.

Also do this in the other tests.